subsys/bluetooth/audio/shell/bap.c

```diff
@@ -2381,69 +2386,109 @@ static uint16_t interval_to_sync_timeout(uint16_t interval)
 
 static bool scan_check_and_sync_broadcast(struct bt_data *data, void *user_data)
 {
-	const struct bt_le_scan_recv_info *info = user_data;
-	char le_addr[BT_ADDR_LE_STR_LEN];
+	struct bt_broadcast_info *sr_info = (struct bt_broadcast_info *)user_data;
 	struct bt_uuid_16 adv_uuid;
-	uint32_t broadcast_id;
 
-	if (data->type != BT_DATA_SVC_DATA16) {
+	switch (data->type) {
+	case BT_DATA_SVC_DATA16:
+		if (data->data_len < BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE) {
+			return true;
+		}
+
+		if (!bt_uuid_create(&adv_uuid.uuid, data->data, BT_UUID_SIZE_16)) {
+			return true;
+		}
+
+		if (bt_uuid_cmp(&adv_uuid.uuid, BT_UUID_BROADCAST_AUDIO) != 0) {
+			return true;
+		}
+
+		sr_info->broadcast_id = sys_get_le24(data->data + BT_UUID_SIZE_16);
 		return true;
-	}
+	case BT_DATA_BROADCAST_NAME:
+		if (!IN_RANGE(data->data_len, BT_AUDIO_BROADCAST_NAME_LEN_MIN,
+			      BT_AUDIO_BROADCAST_NAME_LEN_MAX)) {
+			return false;
+		}
 
-	if (data->data_len < BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE) {
+		memcpy(sr_info->broadcast_name, data->data, data->data_len);
+		sr_info->broadcast_name[data->data_len] = '\0';
+		utf8_trunc(sr_info->broadcast_name);
```

Comment:
```suggestion
		sr_info->broadcast_name[data->data_len] = '\0';
```

Since we are doing a length check we do not need to truncate

Reply:
Ok.